connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -464,10 +465,22 @@ ConfigInfos validateConnectorConfig(Map<String, String> connectorProps, boolean
             connectorProps = worker.configTransformer().transform(connectorProps);
         }
         String connType = connectorProps.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG);
-        if (connType == null)
-            throw new BadRequestException("Connector config " + connectorProps + " contains no connector type");
+        if (connType == null) {
+            return createConnectorClassError("Config contains no connector type");
+        }
+
+        Connector connector;
+        try {
+            connector = getConnector(connType);
+        } catch (ConnectException e) {
+            return createConnectorClassError(e.getMessage());
```

Comment:
I agree. I was tempted to just remove the list of connectors as a whole, but there are other call sites where the exception message is logged. In case it can be useful for diagnostics, I added a parameter instead.